tutorials/incorporating_oli_calculations.ipynb

```diff
@@ -233,45 +299,90 @@
-    "            dbs_file_id,\n",
-    "            water_analysis_apparent_composition,\n",
-    "            survey,\n",
-    "            \"isothermal_composition_survey\",\n",
-    "        )"
+    "# Flash calculation example #1: Water Analysis survey with pH reconciliation.\n",
+    "\n",
+    "# OLIApi is used as a context manager\n",
+    "with OLIApi(credential_manager, interactive_mode=interactive_mode) as oliapi:\n",
+    "    # create a new DBS file\n",
+    "    # alternative thermo_frameworks and databanks are available.\n",
+    "\n",
+    "    # use keep_file = True to save a DBS file ID on the Cloud\n",
+    "    # for use in more than one session\n",
+    "    dbs_file_id = oliapi.generate_dbs_file(\n",
+    "        inflows=inflows,\n",
+    "        thermo_framework=\"MSE (H3O+ ion)\",\n",
+    "        phases=[\"liquid1\", \"solid\"],\n",
+    "        model_name=\"test\",\n",
+    "        keep_file=True,\n",
+    "    )\n",
+    "    # NOTE: liquid2 (non-aqueous) and vapor phases are available but not yet fully supported.\n",
+    "\n",
+    "    # if a DBS file has been retained from a previous session,\n",
+    "    # its flash history and chemistry information can be summarized.\n",
+    "    file_summary = oliapi.get_dbs_file_summary(dbs_file_id)\n",
+    "\n",
+    "    # save chemistry information\n",
+    "    chemistry_info = file_summary[\"chemistry_info\"]\n",
+    "    write_output(chemistry_info[\"result\"], \"chemistry_info\")\n",
+    "\n",
+    "    # create water analysis input\n",
+    "    # Water Analysis uses true species for inputs, i.e., IONS.\n",
+    "    json_input = flash.configure_water_analysis(\n",
+    "        inflows,\n",
+    "        reconciliation=\"ReconcilePh\",\n",
+    "        ph=5,\n",
+    "        allow_solids=True,\n",
+    "        file_name=\"water_analysis_inputs.json\",\n",
+    "    )\n",
+    "    # run Water Analysis flash calculation survey as specified        \n",
+    "    stream_output = flash.run_flash(\n",
+    "        \"wateranalysis\",\n",
+    "        oliapi,\n",
+    "        dbs_file_id,\n",
+    "        json_input,\n",
+    "        survey,\n",
+    "        file_name=\"water_analysis_outputs.json\",\n",
+    "    )\n",
+    "\n",
+    "    # The output of Water Analysis gives apparent species\n",
+    "    # i.e., SALTS, COMPLEXES (MgO, CaO, etc.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "# Flash calculation example #2: isothermal analysis survey\n",
+    "with OLIApi(credential_manager, interactive_mode=False) as oliapi:\n",
+    "\n",
+    "    # Other flash calculations use apparent species for inputs.\n",
+    "\n",
+    "    # These can be obtained in one step with get_apparent_species_from_true(),\n",
+    "    # and saved to a file and reloaded as needed.\n",
+    "    apparent_species = flash.get_apparent_species_from_true(\n",
```

Comment:
@veccp If I skip the problem above with dbs file summary and go straight to these calcs, I get the following error (I also got this error in my own script, which is what made me look back to this tutorial in its latest state to see if it was my doing or not): 
![image](https://github.com/watertap-org/watertap/assets/22691844/d3bce76e-4538-493d-869a-2553141cea86)

Reply:
Actually- this could still be my doing! I forgot to mention that I merged this PR with #1303 on a local branch and quickly tried to resolve merge conflicts while maintaining the key changes from both PRs. So I probably should just check this tutorial based on this PR's changes only....

Reply:
@adam-a-a Thanks for the comments. There were a few bugs in the tutorial that I just fixed. It should work as is